src/queues/UnlistedUserRedactionQueue.ts

```diff
@@ -13,12 +13,15 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
-
 import { extractRequestError, LogLevel, LogService, MatrixClient, Permalinks } from "matrix-bot-sdk";
 import { logMessage } from "../LogProxy";
 import config from "../config";
 
-export class AutomaticRedactionQueue {
+/**
+ * This is used to redact new events from users who are not banned from a watched list, but have been flagged
```

Comment:
It's really a queue of users who have been spotted by the FirstMessageIsImage/Flood protection in a room (if someone posts their first message as an image then mjolnir will ban them from that room, add them to this queue and if they start sending messages in other rooms, mjolnir will redact them by checking the event sender against this "queue" ). Of course that means that the user isn't banned from any list mjolnir is watching (or they wouldn't be able to post new messages), I think it's intended to buy time and be spotted by moderators so they can make a decision on whether the user is really bad or not.

Reply:
Then maybe `RoomProtectionRedactionQueue`?

```js
/**
 * A queue of events spotted by one of the room protections (e.g. first message is image, flood protection)
 * and marked to be redacted asynchronously.
 *
 * <more details>
 */
```

Reply:
No, this is not a queue of events, it's a queue of users for redacting events that they have not sent yet. It's not touched by the scope of this PR, it just needs renaming because the names sound too similar and their function is really different. I have tried to explain that in the descriptions since which is not shown in the diff.